src/Microsoft.Data.Analysis/PrimitiveDataFrameColumn.BinaryOperations.cs

```diff
@@ -16,33 +16,33 @@ public partial class PrimitiveDataFrameColumn<T> : DataFrameColumn
     {
         public override DataFrameColumn Add(DataFrameColumn column, bool inPlace = false)
         {
-            switch (column)
+            switch (column.DataType)
             {
-                case PrimitiveDataFrameColumn<bool> boolColumn:
+                case Type boolType when boolType == typeof(bool):
```

Comment:
I think your existing pattern matching will work correctly here. Why this change?

Also, for the existing code, you don't need the additional cast below. You can just say:

```C#
            switch (column)
            {
                case PrimitiveDataFrameColumn<bool> boolColumn:
                    return AddImplementation(boolColumn, inPlace);
```

Reply:
Right, I thought the same thing. But, we have new column types now, so we need to test against the datatype it holds instead of the column type itself. So `IntDataFrameColumn` and `PrimitiveDataFrameColumn<int>` would both work now

Reply:
I tested this in an isolated environment, and it works:

```C#
    class Primitive<T> where T : struct
    {
    }
    class Int32Column : Primitive<int>
    {
    }
    class Program
    {
        static void Main(string[] args)
        {
            Print(new Int32Column());
            Print(new Primitive<int>());
            Print(new Primitive<bool>());
        }
        private static void Print(object o)
        {
            switch (o)
            {
                case Primitive<int> intColumn:
                    Console.WriteLine("Got int");
                    break;
                case Primitive<long> intColumn:
                    Console.WriteLine("Got long");
                    break;
                default:
                    Console.WriteLine("got nothing");
                    break;
            }
        }
    }
```

prints:
```
Got int
Got int
got nothing
```